### PR TITLE
Fixed non-persisted identity/account lookups

### PR DESCRIPTION
**Pull request by krzlabrdx**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs [hunk 1]
```diff
@@ -145,32 +145,32 @@ private record struct ExplicitMetadataLookup(long EntityId, string MetadataKey);
...
-        var royaltyVaultsBalance = componentEntities.Any() && (optIns.ComponentRoyaltyVaultBalance || optIns.PackageRoyaltyVaultBalance)
-            ? await RoyaltyVaultBalance(componentEntities.Select(x => x.Id).ToArray(), ledgerState, token)
+        var royaltyVaultsBalance = globalPersistedComponentEntities.Any() && (optIns.ComponentRoyaltyVaultBalance || optIns.PackageRoyaltyVaultBalance)
+            ? await RoyaltyVaultBalance(globalPersistedComponentEntities.Select(x => x.Id).ToArray(), ledgerState, token)
             : null;
 
         var fungibleResources =
-            await EntityFungibleResourcesPageSlice(globalComponentEntities.Select(e => e.Id).ToArray(), aggregatePerVault, 0, _endpointConfiguration.Value.DefaultPageSize, ledgerState, token);
-        var nonFungibleResources = await EntityNonFungibleResourcesPageSlice(globalComponentEntities.Select(e => e.Id).ToArray(), aggregatePerVault, optIns.NonFungibleIncludeNfids, 0,
+            await EntityFungibleResourcesPageSlice(componentEntities.Select(e => e.Id).ToArray(), aggregatePerVault, 0, _endpointConfiguration.Value.DefaultPageSize, ledgerState, token);
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:172]: I don't know how this affects the result, but I thought we wanted to return something for all global entries (instantiated or no)? Or are we adding it back to internal entities too?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs [hunk 2]
```diff
@@ -145,32 +145,32 @@ private record struct ExplicitMetadataLookup(long EntityId, string MetadataKey);
...
         var fungibleVaultEntities = entities.OfType<InternalFungibleVaultEntity>().ToList();
         var nonFungibleVaultEntities = entities.OfType<InternalNonFungibleVaultEntity>().ToList();
+        var globalPersistedComponentEntities = componentEntities.Where(x => x.IsGlobal && x.Id != default).ToList();
 
         // TODO ideally we'd like to run those in parallel
         var metadata = await GetMetadataSlices(entities.Select(e => e.Id).ToArray(), 0, _endpointConfiguration.Value.DefaultPageSize, ledgerState, token);
-        var roleAssignmentsHistory = await _roleAssignmentQuerier.GetRoleAssignmentsHistory(globalComponentEntities, ledgerState, token);
-        var stateHistory = await GetStateHistory(componentEntities, ledgerState, token);
+        var roleAssignmentsHistory = await _roleAssignmentQuerier.GetRoleAssignmentsHistory(globalPersistedComponentEntities, ledgerState, token);
+        var stateHistory = await GetStateHistory(globalPersistedComponentEntities, ledgerState, token);
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:157]: I imagine we should return state for internal components? So this should be persistedComponentEntities, not globalPersistedComponentEntities?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs [hunk 3]
```diff
@@ -145,32 +145,32 @@ private record struct ExplicitMetadataLookup(long EntityId, string MetadataKey);
...
         var packageEntities = entities.OfType<GlobalPackageEntity>().ToList();
         var fungibleVaultEntities = entities.OfType<InternalFungibleVaultEntity>().ToList();
         var nonFungibleVaultEntities = entities.OfType<InternalNonFungibleVaultEntity>().ToList();
+        var globalPersistedComponentEntities = componentEntities.Where(x => x.IsGlobal && x.Id != default).ToList();
 
         // TODO ideally we'd like to run those in parallel
         var metadata = await GetMetadataSlices(entities.Select(e => e.Id).ToArray(), 0, _endpointConfiguration.Value.DefaultPageSize, ledgerState, token);
-        var roleAssignmentsHistory = await _roleAssignmentQuerier.GetRoleAssignmentsHistory(globalComponentEntities, ledgerState, token);
-        var stateHistory = await GetStateHistory(componentEntities, ledgerState, token);
+        var roleAssignmentsHistory = await _roleAssignmentQuerier.GetRoleAssignmentsHistory(globalPersistedComponentEntities, ledgerState, token);
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:156]: We want to return default role assignments for pre-allocated, uninstantiated components too. I don't know if this affects that?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs [hunk 4]
```diff
@@ -145,32 +145,32 @@ private record struct ExplicitMetadataLookup(long EntityId, string MetadataKey);
     {
         var entities = await GetEntities(addresses, ledgerState, token);
         var componentEntities = entities.OfType<ComponentEntity>().ToList();
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:147]: Are accounts a subclass of ComponentEntity?

**Review by dhedey — COMMENTED** [review]
I'm not sure whether I'm reading the queries here, or the model creation.

If I'm reading the queries, then lots of my comments may not be relevant because we don't need to query for non-instantiated data. But we do need to be including models for pre-allocated un-instanted accounts and identities.

Might be worth a call to run through this?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs [hunk 5]
```diff
@@ -145,32 +145,32 @@ private record struct ExplicitMetadataLookup(long EntityId, string MetadataKey);
...
 
         var resourcesSupplyData = await GetResourcesSupplyData(resourceEntities.Select(x => x.Id).ToArray(), ledgerState, token);
         var packageBlueprintHistory = await GetPackageBlueprintHistory(packageEntities.Select(e => e.Id).ToArray(), ledgerState, token);
         var packageCodeHistory = await GetPackageCodeHistory(packageEntities.Select(e => e.Id).ToArray(), ledgerState, token);
         var packageSchemaHistory = await GetPackageSchemaHistory(packageEntities.Select(e => e.Id).ToArray(), ledgerState, token);
         var fungibleVaultsHistory = await GetFungibleVaultsHistory(fungibleVaultEntities, ledgerState, token);
         var nonFungibleVaultsHistory = await GetNonFungibleVaultsHistory(nonFungibleVaultEntities, optIns.NonFungibleIncludeNfids, ledgerState, token);
-        var correlatedAddresses = await GetCorrelatedEntityAddresses(entities, componentEntities, packageBlueprintHistory, ledgerState, token);
+        var correlatedAddresses = await GetCorrelatedEntityAddresses(entities, globalPersistedComponentEntities, packageBlueprintHistory, ledgerState, token);
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:165]: I think this is incorrect. It should still be componentAddresses, or maybe more specifically, `instantiatedComponentAddresses`

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs (hunk 1 above)
Comment (dhedey) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:172]: After our call - can we add a comment that, for backwards compatibility, we're returning fungible/non-fungible resources for _all_ components, even non-global components.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs (hunk 3 above)
Comment (dhedey) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:156]: After call:
* This looks correct.
* Can we add a comment to each of these, explaining where the "virtualness" is handled.

**Review by dhedey — CHANGES_REQUESTED** [review]
Looks good subject to agreed changes

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs (hunk 5 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs]: Fixed.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs (hunk 3 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:156]: Added necessary comment(s).

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs (hunk 1 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:173]: Added necessary comment(s).

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs (hunk 3 above)
Comment (dhedey) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:156]: Minor, but we're missing a comment here explaining that role assignments for virtual entities are handled below the switch statement.

**Review by dhedey — APPROVED** [review]
Looks good, thanks. Happy for you to merge with or without the additional comment

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs (hunk 3 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:156]: Added extra comment

**Pull request merged**